app/javascript/packages/document-capture/components/form-error-message.jsx

```diff
@@ -23,16 +24,31 @@ const NBSP_UNICODE = '\u00A0';
  */
 export class RequiredValueMissingError extends Error {}
 
+/**
+ * An error representing user declined access to camera.
+ */
+export class CameraAccessDeclinedError extends Error {}
+
 /**
  * @param {FormErrorMessageProps} props Props object.
  */
-function FormErrorMessage({ error }) {
+function FormErrorMessage({ error, isDetail = false }) {
```

Comment:
what if we just made a separate detail error message like `CameraAccessDeclinedDetailError` ?

Reply:
> what if we just made a separate detail error message like `CameraAccessDeclinedDetailError` ?

I considered this idea, and the reason I did it this way is because it felt to me that they are essentially one and the same, so having two different error classes felt unnecessary. It also matched well with the translation key pair names, and I could imagine some of the other errors having "expanded" forms if we'd similarly want to show a top-level error.

On the other hand, the current implementation for `FormSteps` requires that they be [thrown as separate errors](https://github.com/18F/identity-idp/blob/1ba50cad137eabf6bd99bd4a1f98a52081a7360a/app/javascript/packages/document-capture/components/documents-step.jsx#L89-L90). And some of the other error types like `UploadFormEntryError` are sort of treated as "detail" by default.

I don't have a strong attachment one way or the other. LMK if you have a preference. Otherwise I'll probably just keep it as-is.

Reply:
yeah throwing multiple errors surprised me, my other thought was what about adding a property to the error itself?

But I guess this wouldn't necessarily solve that. It seemed weird to me to add this extra option when we only use it in one case, but it's not that big of a deal